app/jobs/chat_channels/index_job.rb

```diff
@@ -5,10 +5,7 @@ class IndexJob < ApplicationJob
     queue_as :chat_channels_index
 
     def perform(chat_channel_id:)
-      chat_channel = ChatChannel.find_by(id: chat_channel_id)
-
-      return unless chat_channel
-
+      chat_channel = ChatChannel.find(chat_channel_id)
```

Comment:
Please go back to the previous code, if you look at other jobs we always exit the job without raising an exception for 404.

The reason being that jobs are retried and there's no possible positive outcome in retrying a job if the row in the table is not there